clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/StoreRepushCommand.java

```diff
@@ -88,14 +109,106 @@ public List<String> getShellCmd() {
   }
 
   private void processOutput(String output, int exitCode) {
-    result.setStdOut(output);
-    result.setExitCode(exitCode);
-    result.parseStandardOutput();
-    result.setCoreWorkDone(true);
+    getResult().setStdOut(output);
+    getResult().setExitCode(exitCode);
+    getResult().parseStandardOutput();
+    getResult().setCoreWorkDone(true);
+  }
+
+  public ControllerClient buildControllerClient(String clusterName, String url, Optional<SSLFactory> sslFactory) {
+    return new ControllerClient(clusterName, url, sslFactory);
+  }
+
+  public Pair<Boolean, String> getRepushViability() {
+    String url = getParams().getUrl();
+    ControllerClient cli = getParams().getPCtrlCliWithoutCluster();
+    LocalDateTime timestamp = getParams().getTimestamp();
+    String destFabric = getParams().getDestFabric();
+    String s = getParams().getStore();
+    try {
+      String clusterName = cli.discoverCluster(s).getCluster();
+      if (clusterName == null) {
+        return Pair.of(false, "unable to discover cluster for store (likely invalid store name)");
```

Comment:
It looks like a good fit for using a enum class to wrap around this information in one place, something like com.linkedin.venice.Arg, instead of creating them one by one. Not a strong opinion and we can do it in a future code cleanup.